test/DryIoc.IssuesTests/GHIssue48_Ability_to_add_instances_into_scope_in_OpenScope_instead_of_UseInstance.cs

```diff
@@ -12,14 +12,13 @@ public void Add_one_instance_to_scope()
             container.Register<A>();
 
             var b = new B();
-            using (var scope = container.OpenScopeWithPresetServices(instances: b))
+            using (var scope = container.OpenScopeWithPresetServices("ScopeName", instances: b))
```

Comment:
@LunicLynx I am not sure what that suppose to fix. It is just saying that scope `name` will cache expression with injected instance in different bucket from default path, thus allowing the default path to correctly resolve/create the expression after scope is gone. But it does not solve the default scope and default path caching problem.

Reply:
The result expression still will look like this `r => new A(b)`, where `b` is `Expression.Constant(b)`.
So if you let's say change `b` to `b1`, the cached expression still list the `b`.

I am proposing to dynamically inject instances into the expression like so `r => new A(r.Resolve<B>(...))`.
Then inside the `r.Resolve<B>(...)` container will check the scope or singletons (when no scope is available) for the presence of `B` what instance it may be. Ann when no instances are found, it fallbacks to normal registration search for `B`.

Reply:
Yes i guess i do not understand what the intention of that test was. I don't actually know how to test this case with the current implementation. I would introduce an interface around the delegate creation. The mock it for this case an count the invocations. Expecting it to be one.